activerecord/lib/active_record/associations.rb

```diff
@@ -1676,6 +1676,11 @@ def has_one(name, scope = nil, **options)
         #   belongs_to :note, query_constraints: [:organization_id, :note_id]
         def belongs_to(name, scope = nil, **options)
           reflection = Builder::BelongsTo.build(self, name, scope, options)
+
+          if self.ignored_columns.include?(reflection.foreign_key.to_s)
+            raise ArgumentError, "You tried to define a belongs_to association for #{name}, but the column #{reflection.foreign_key} is ignored. You can remove it from the ignored_columns list if you want to use it."
+          end
+
```

Comment:
For the super class Builder we raise an ArgumentError in the `build` method:
https://github.com/rails/rails/blob/9c41f5828cd4134469ee57a2a3b523a27f93a630/activerecord/lib/active_record/associations/builder/association.rb#L25-L30

Maybe override `Builder::BelongsTo.build` to check the arguments there?

Reply:
thanks @p8 for the suggestion, let me check.

Reply:
@p8 refactored as per suggestion.